File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int m = nums[0], c = 1;
        for(int i=1; i<nums.size(); i++){
            if(nums[i]==m)c++;
            else{
                if(c!=0)c--;
                else{
                    m=nums[i];
                    c=1;
                }
            }
        }
        return m;
    }
};
// Time complexity O(n)
// space complexity O(1)

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let cand = nums[0];
    let c = 1;
    for(let i=1; i<nums.length; i++){
        if(nums[i]===cand)c++;
        else c--;

        if(c<0){
            cand = nums[i];
            c=1;
        }
    }
    return cand;
};